### PR TITLE
FIX: mask now pulls from loaded calibration stream

### LINKED ISSUE #128: BUG: check that mask inherits from geometry union.

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes #128

### CODE REVIEW

**Comment by CJ-Wright** [comment]
This is pending https://github.com/xpdAcq/SHED/pull/84#pullrequestreview-60792063

File: xpdan/calib.py [hunk 1]
```diff
@@ -170,12 +170,16 @@ def img_calibration(img, wavelength, calibrant=None,
     pyFAI documentation:
     http://pyfai.readthedocs.io/en/latest/
     """
+    wavelength *= 10**-10
     if detector is None:
         detector = 'perkin_elmer'
     if calibrant is None:
         calibrant = 'Ni'
+    elif isinstance(calibrant, list):
```
Comment (CJ-Wright) [inline xpdan/calib.py]: Pull calibrant from dspacings

File: xpdan/db_utils.py [hunk 2]
```diff
@@ -147,17 +147,25 @@ def query_dark(db, docs, schema=1):
 def query_background(db, docs, schema=1):
     if schema == 1:
         doc = docs[0]
-        return db(sample_name=doc['bkgd_sample_name'],
-                  is_dark={'$exists': False})
+        sample_name = doc.get('bkgd_sample_name')
```
Comment (CJ-Wright) [inline xpdan/db_utils.py]: Be a bit more graceful about pulling the background from the document.

File: xpdan/db_utils.py [hunk 3]
```diff
@@ -147,17 +147,25 @@ def query_dark(db, docs, schema=1):
...
+                      is_dark={'$exists': False})
+        else:
+            return []
 
 
 def temporal_prox(res, docs):
     # If there is only one result just use that one
     if isinstance(res, Header):
-        return res
+        return [res]
```
Comment (CJ-Wright) [inline xpdan/db_utils.py]: Always return an iterable

**Comment by CJ-Wright** [comment]
Need to convert tth radians into degrees before doing anything else.

File: xpdan/calib.py [hunk 4]
```diff
@@ -170,12 +170,16 @@ def img_calibration(img, wavelength, calibrant=None,
...
+        calibrant = Calibrant(dSpacing=calibrant, wavelength=wavelength)
     # configure calibration instance
-    c = Calibration(calibrant, detector, wavelength)
+    c = Calibration(calibrant=calibrant, detector=detector,
+                    wavelength=wavelength)
```
Comment (chiahaoliu) [inline xpdan/calib.py]: 👍

File: xpdan/db_utils.py (hunk 2 above)
Comment (chiahaoliu) [reply xpdan/db_utils.py]: 👍

**Pull request merged**